api/graphql/notificationQuerySet.js

```diff
@@ -0,0 +1,37 @@
+import presentQuerySet from '../../lib/graphql-bookshelf-bridge/util/presentQuerySet'
+
+const userColumns = q => q.column('users.id', 'name', 'avatar_url')
+
+export default function notificationQuerySet (community, userId, options) {
+  return Notification.query(q => {
+    q.where('reader_id', userId)
+    .leftJoin('activities', function () {
+      this.on('activities.id', '=', 'notifications.activity_id')
+    })
+    q.where('medium', Notification.MEDIUM.InApp)
+    q.limit(options.first)
+    q.offset(options.offset)
+    q.select(bookshelf.knex.raw('notifications.*, count(*) over () as total'))
+    q.orderBy('activities.created_at', 'desc')
+
+    Activity.filterInactiveContent(q)
+    if (community) Activity.joinWithCommunity(community.id, q)
+  })
+}
+
+export function fetchNotificationQuerySet (community, userId, options) {
+  return notificationQuerySet(community, userId, options).fetchAll({withRelated: [
```

Comment:
Basically you shouldn't need to use a getter and an explicit query set, because you don't need to search notifications. It can be a relation. That will simplify some of this dramatically. But then you do lose the ability to use the syntatic sugar methods to skip having to go through `activity` explicitly to get to `post` or `comment`. I think that's probably for the best in order to avoid this gargantuan `fetchAll`.